### PR TITLE
BC-2697 - fix publish task

### PR DESCRIPTION
**Pull request by WojciechGrancow**
# Short Description

## Links to Ticket and related Pull-Requests
https://ticketsystem.dbildungscloud.de/browse/BC-2697

## Changes
As part of the repair, a time delay has been added to change the status from planned to running. Unfortunately, some windows users were experiencing differences of a few seconds in the publication of a task, resulting in incorrect UI behaviour

On other systems i.e. linux, mac it was not possible to reproduce this error.

## Screenshots of UI changes

In ticket https://ticketsystem.dbildungscloud.de/browse/BC-2697 in the comments are short videos showing the application working correctly in three browsers chrome, edge, firefox on win10

## Checklist before merging

- [x] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [x] PO: Any deviation from requirements was agreed with Product-Owner / ticket author / support-team

> Notice: Please keep this Pull-Request as a Draft (or add WIP label), until it is ready to be reviewed

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -58,6 +58,7 @@
 		"material-icons-svg": "^1.0.0",
 		"mobile-drag-drop": "^2.3.0-rc.2",
 		"moment": "^2.29.1",
+		"node-gyp": "^9.3.0",
```
Comment (muratmerdoglu-dp) [inline package.json:61]: Is this dependency really required?

File: src/components/molecules/RoomTaskCard.vue [hunk 2]
```diff
@@ -172,7 +172,11 @@ export default {
 		},
 		isPlanned() {
 			const scheduledDate = this.task.availableDate;
-			return scheduledDate && new Date(scheduledDate) > new Date();
+			const delay = 30 * 1000;
+			return (
+				scheduledDate &&
+				new Date(new Date(scheduledDate).getTime() - delay) > new Date()
```
Comment (muratmerdoglu-dp) [inline src/components/molecules/RoomTaskCard.vue:178]: Does the error occur only after the `publish(isPlanned)` button clicked? Have you tested the `finish` and `revert` functions?

Comment (WojciechGrancow) [reply src/components/molecules/RoomTaskCard.vue:178]: yes, this error occur only when task was switched from draft to publish.
With finish and revert functions behaviour of app (UI) is OK

File: package.json (hunk 1 above)
Comment (uidp) [reply package.json:61]: Don't think we need that to solve the issue, right?

Comment (WojciechGrancow) [reply package.json]: this dependecny is required on my local environment, so I make correction in PR

File: src/components/molecules/RoomTaskCard.vue (hunk 2 above)
Comment (WojciechGrancow) [reply src/components/molecules/RoomTaskCard.vue:178]: ok i will check it but, on the branch, becouse it is not possible to reproduce on local env

**Review by muratmerdoglu-dp — COMMENTED** [review]
The package-lock.json file should be removed!

File: src/components/molecules/RoomTaskCard.unit.ts [hunk 3]
```diff
@@ -492,6 +492,64 @@ describe("@components/molecules/RoomTaskCard", () => {
...
+						description: "some description here",
+					},
+				};
+				const wrapper = getWrapper({ ...localProps, role });
+				const { vm } = wrapper;
+				expect(vm.isPlanned).toBe(false);
+			});
+
+			it("should return true value after calculated isPlanned() method", () => {
+				const inFutureDate = new Date(Date.now() + 3600 * 1000 * 24);
```
Comment (muratmerdoglu-dp) [inline src/components/molecules/RoomTaskCard.unit.ts:525]:
```suggestion
				jest.useFakeTimers("modern").setSystemTime(new Date()); // this line sets a permanent fake time
				const inFutureDate = new Date(Date.now() + 5001);
```

File: package-lock.json [hunk 4]
```diff
@@ -27,6 +27,7 @@
 				"material-icons-svg": "^1.0.0",
 				"mobile-drag-drop": "^2.3.0-rc.2",
 				"moment": "^2.29.1",
+				"node-gyp": "^9.3.0",
```
Comment (muratmerdoglu-dp) [inline package-lock.json:30]: This `package-lock.json` file should be removed.

**Review by muratmerdoglu-dp — CHANGES_REQUESTED** [review]

File: package-lock.json (hunk 4 above)
Comment (WojciechGrancow) [reply package-lock.json]: packed-lock removed form commited changes

**Review by muratmerdoglu-dp — APPROVED** [review]

**Pull request merged**